### DISCUSSION TITLE
Does numberOfRuns create a new context for each run?

### DISCUSSION THREAD

**Discussion by zshnr** [GoogleChrome/lighthouse-ci#546] [Q&A] [answered]
Hello! 

I went through the code before I posted this just to double check my understanding. The way I read the lighthouse-ci code is that it creates one instance of puppeteer and one instance of the browser inside puppeteer. And in that instance, runs the benchmark either once or as many times as specified by the numberOfRuns property. Is this correct?

The reason I ask is for our use case it would've been ideal if each run was a separate browser context (ideally incognito). 

It is ok if that is not the case, I might end up writing our own puppeteer script to collect results and upload to our internal CI server then. But I just wanted to be sure before I went down that route.

Thanks a lot!

**Comment by patrickhulce** [comment, marked as answer]
> The way I read the lighthouse-ci code is that it creates one instance of puppeteer and one instance of the browser inside puppeteer. And in that instance, runs the benchmark either once or as many times as specified by the numberOfRuns property. Is this correct?

Yes that's correct.

> The reason I ask is for our use case it would've been ideal if each run was a separate browser context (ideally incognito).

Can your use case be satisfied by clearing storage for the origin using the DevTools protocol instead? Lighthouse does this by default but only for a few types of storage. If you want a clean slate like incognito you could instead invoke [Storage.clearDataForOrigin](https://chromedevtools.github.io/devtools-protocol/tot/Storage/#method-clearDataForOrigin) in a LHCI puppeteer script for anything you want to reset in between runs?

**Reply by zshnr** [reply]
OMG that actually might work for us! 

Thank you so much for answering! :D
